janus_server/src/task.rs

```diff
@@ -137,6 +137,44 @@ enum VdafSerialization {
     FakeFailsPrepStep,
 }
 
+/// A verification key for a VDAF, with a variable length. It must be kept secret from clients to
+/// maintain robustness, and it must be shared between aggregators.
+#[derive(Clone, PartialEq, Eq)]
+pub struct VerifyKeyDynamicSize(Vec<u8>);
```

Comment:
I don't like that we have two types for verification keys based on whether we happen to know the expected size of the verification key. The names are also quite a mouthful. Also, `VerifyKeyDynamicSize` will never actually be used as a verification key -- using it requires converting to a `VerifyKeyStaticSize` first, since all users of verification keys are generic over the length of the verify key.

How about we replace `VerifyKeyDynamicSize` by a purpose-generic type called e.g. `SecretBytes` whose purpose is to hold sensitive bytes of unknown purpose. Then rename `VerifyKeyStaticSize` to `VerifyKey`.

This IMO improves the naming a lot. A downside is that we no longer know the purpose of an arbitrary `SecretBytes` value. However, I think this downside is mitigated by the fact that we are storing these values in fields whose names describe their purpose, and always converting these values into a "purposeful" type (`VerifyKey`) before use -- once `Task` fields are private, I think the verify key getter should return a `VerifyKey` by performing the conversion as necessary, with the fact that the bytes are stored as `SecretBytes` inside the struct being an implementation detail. WDYT?